tsdb/agent/db_test.go

```diff
@@ -879,6 +882,61 @@ func TestDBAllowOOOSamples(t *testing.T) {
 	require.NoError(t, db.Close())
 }
 
+func TestDBOutOfOrderTimeWindow(t *testing.T) {
+	tc := []struct {
+		outOfOrderTimeWindow, firstTs, secondTs int64
+		reject                                  bool
+	}{
+		{0, 100, 101, false},
+		{0, 100, 100, true},
+		{0, 100, 99, true},
+		{100, 100, 1, false},
+		{100, 100, 0, true},
+	}
+
+	for _, c := range tc {
+		t.Run(fmt.Sprintf("outOfOrderTimeWindow=%d, firstTs=%d, secondTs=%d, reject=%t", c.outOfOrderTimeWindow, c.firstTs, c.secondTs, c.reject), func(t *testing.T) {
+			reg := prometheus.NewRegistry()
+			opts := DefaultOptions()
+			opts.SetOutOfOrderTimeWindow(c.outOfOrderTimeWindow)
+			s := createTestAgentDB(t, reg, opts)
+			app := s.Appender(context.TODO())
+
+			var expectedErr error
+			if c.reject {
+				expectedErr = storage.ErrOutOfOrderSample
+			}
```

Comment:
What do you think about changing `reject bool` to `error`, that could be either unset(nil) or `storage.ErrOutOfOrderSample` when declaring the test cases?

This way we could delete this if, just a little less code to read 😛

Reply:
We wouldn't be able to delete the if below, but that's fine I guess

Reply:
Makes sense. Done!